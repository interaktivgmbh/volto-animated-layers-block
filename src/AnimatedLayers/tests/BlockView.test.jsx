import React from 'react';
import renderer from 'react-test-renderer';
import AnimatedLayersBlockView from '../BlockView';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import config from '@plone/volto/registry';
import { MemoryRouter } from 'react-router-dom';
import AnimatedLayersBlockEdit from '../BlockEdit';

const mockStore = configureStore();

//noinspection JSUnresolvedFunction
beforeAll(() => {
  config.blocks.blocksConfig = {
    ...config.blocks.blocksConfig,

    animatedlayers: {
      id: 'animatedlayers',
    },
  };
});

describe('View', () => {
  const store = mockStore({
    intl: {
      locale: 'en',
      messages: {},
    },
  });

  it('renders AnimatedLayers View component without data', () => {
    const data = {
      '@type': 'animatedlayers',
      AnimatedLayers: [],
    };

    const component = renderer.create(
      <Provider store={store}>
        <AnimatedLayersBlockView data={data} />
      </Provider>,
    );
    const json = component.toJSON();
    //noinspection JSUnresolvedFunction
    expect(json).toMatchSnapshot();
  });

  it('renders AnimatedLayers View component with data', () => {
    const data = {
      '@type': 'animatedlayers',
      title: 'Test-Title',
      description: 'Test-Description',
      layers: [
        {
          label: 'Layer 1',
          visible: true,
          size: '0.5',
          top: '10',
          left: '20',
          animation: [{ '@id': 'https://test.com/test.svg' }],
        },
        {
          label: 'Layer 2',
          visible: true,
          size: '1',
          top: '0',
          left: '50',
          animation: { '@id': '/test.svg' },
        },
      ],
      buttons: [
        {
          label: 'Layer 1',
          link: [{ '@id': 'https://test.com/test.svg' }],
        },
        {
          label: 'Layer 2',
          link: { '@id': '/test.svg' },
        },
      ],
    };

    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <AnimatedLayersBlockEdit data={data} />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    //noinspection JSUnresolvedFunction
    expect(json).toMatchSnapshot();
  });
});
