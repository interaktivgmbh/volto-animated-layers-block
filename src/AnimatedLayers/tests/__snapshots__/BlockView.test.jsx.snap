// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View renders AnimatedLayers View component with data 1`] = `
<div
  className="animated-layers-block"
>
  <div
    className="grid-layer"
  />
  <div
    className="columns"
  >
    <div
      className="column layers-selection"
    >
      <div
        className="block-title"
      >
        Test-Title
      </div>
      <div
        className="block-description"
      >
        Test-Description
      </div>
      <div
        className="layers"
      >
        <div
          className="layer"
        >
          <label
            className="toggler-wrapper style-1"
          >
            <input
              checked={true}
              className="layer-selector"
              data-label="Layer 1"
              onChange={[Function]}
              type="checkbox"
            />
            <div
              className="toggler-slider"
            >
              <div
                className="toggler-knob"
              />
            </div>
          </label>
          <div
            className="layer-text"
          >
            Layer 1
          </div>
        </div>
        <div
          className="layer"
        >
          <label
            className="toggler-wrapper style-1"
          >
            <input
              checked={true}
              className="layer-selector"
              data-label="Layer 2"
              onChange={[Function]}
              type="checkbox"
            />
            <div
              className="toggler-slider"
            >
              <div
                className="toggler-knob"
              />
            </div>
          </label>
          <div
            className="layer-text"
          >
            Layer 2
          </div>
        </div>
      </div>
      <div
        className="buttons-wrapper"
      >
        <a
          className="btn btn-primary"
          href="https://test.com/test.svg"
          rel="noreferrer"
          target="_blank"
        >
          Layer 1
        </a>
        <a
          className="btn btn-primary"
          href="/test.svg"
          onClick={[Function]}
        >
          Layer 2
        </a>
      </div>
    </div>
    <div
      className="column layers-result"
    >
      <div
        className="layers"
      >
        <div
          className="layer"
          data-layer="layer-0"
          style={
            Object {
              "left": "20%",
              "opacity": "1",
              "top": "10%",
              "transform": "scale(0.5)",
            }
          }
        >
          <object
            data="https://test.com/test.svg"
            type="image/svg+xml"
          />
        </div>
        <div
          className="layer"
          data-layer="layer-1"
          style={
            Object {
              "left": "50%",
              "opacity": "1",
              "top": "0%",
              "transform": "scale(1)",
            }
          }
        >
          <object
            data="/++api++//test.svg/@@display-file-svg"
            type="image/svg+xml"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`View renders AnimatedLayers View component without data 1`] = `
<div
  className="animated-layers-block"
>
  <div
    className="grid-layer"
  />
  <div
    className="columns"
  >
    <div
      className="column layers-selection"
    >
      <div
        className="block-title"
      />
      <div
        className="block-description"
      />
      <div
        className="layers"
      />
      <div
        className="buttons-wrapper"
      />
    </div>
    <div
      className="column layers-result"
    >
      <div
        className="layers"
      />
    </div>
  </div>
</div>
`;
